src/ffi/credential.rs

```diff
@@ -228,3 +196,446 @@ pub extern "C" fn anoncreds_credential_get_attribute(
+///
+/// # Params
+/// cred:       object handle pointing to W3C AnonCreds credential
+/// proof:      data integrity proof as JSON string
+/// cred_p:     reference that will contain update credential
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_w3c_credential_add_non_anoncreds_integrity_proof(
+    cred: ObjectHandle,
+    proof: FfiStr,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {
+    catch_error(|| {
+        check_useful_c_ptr!(cred_p);
+
+        let proof = proof
+            .as_opt_str()
+            .ok_or_else(|| err_msg!("Missing proof"))?;
+        let proof: NonAnonCredsDataIntegrityProof =
+            serde_json::from_str(proof).map_err(|_| err_msg!("Unable to parse proof"))?;
+
+        let mut cred = cred.load()?.cast_ref::<W3CCredential>()?.clone();
+
+        cred.add_proof(CredentialProof::NonAnonCredsDataIntegrityProof(proof));
+
+        let cred = ObjectHandle::create(cred)?;
+        unsafe { *cred_p = cred };
+
+        Ok(())
+    })
+}
+
+/// Set id to W3C AnonCreds credential
+///
+/// # Params
+/// cred:       object handle pointing to W3C AnonCreds credential
+/// id:         id to add into credential
+/// cred_p:     reference that will contain update credential
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_w3c_set_credential_id(
+    cred: ObjectHandle,
+    id: FfiStr,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {
+    catch_error(|| {
+        check_useful_c_ptr!(cred_p);
+
+        let id = id.as_opt_str().ok_or_else(|| err_msg!("Missing id"))?;
+
+        let mut cred = cred.load()?.cast_ref::<W3CCredential>()?.clone();
+
+        cred.set_id(URI::from(id));
+
+        let cred = ObjectHandle::create(cred)?;
+        unsafe { *cred_p = cred };
+
+        Ok(())
+    })
+}
+
+/// Set subject id to W3C AnonCreds credential
+///
+/// # Params
+/// cred:       object handle pointing to W3C AnonCreds credential
+/// id:         subject id to add into credential
+/// cred_p:     reference that will contain update credential
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_credential_w3c_subject_id(
+    cred: ObjectHandle,
+    id: FfiStr,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {
+    catch_error(|| {
+        check_useful_c_ptr!(cred_p);
+
+        let id = id.as_opt_str().ok_or_else(|| err_msg!("Missing id"))?;
+
+        let mut cred = cred.load()?.cast_ref::<W3CCredential>()?.clone();
+
+        cred.set_subject_id(URI::from(id));
+
+        let cred = ObjectHandle::create(cred)?;
+        unsafe { *cred_p = cred };
+
+        Ok(())
+    })
+}
+
+/// Add context to W3C AnonCreds credential
+///
+/// # Params
+/// cred:       object handle pointing to W3C AnonCreds credential
+/// context:    context to add into credential
+/// cred_p:     reference that will contain update credential
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_w3c_credential_add_context(
```

Comment:
I do not have strong vision and opinion regarding these methods. It may be clear during the integration of anoncreds-rs into framework. In general, the goal of adding these methods was supporting the case when an additional NonAnonCreds identity proof needs to be added.
The flow would be issue w3c credential -> do non-anoncreds signature on the application level -> append signature to the object -> serialize and store. 
If you consider these methods as redundant, we can drop them.

Reply:
Please see my comments on #273, there's some relation to this comment.

@swcurran might be good for the WG call later today